### PR TITLE
update ListUsers type filter to array

### PR DESCRIPTION
**Pull request by cyinma**
its likely that we will add additional user type in future and it will be much more useful if the type filter is an array. this is newly added and unused yet, so I am going to just change it in place

### CODE REVIEW

File: proto/buf/alpha/registry/v1alpha1/user.proto [hunk 1]
```diff
@@ -119,8 +119,8 @@ message ListUsersRequest {
   bool reverse = 3;
   // If the user_state_filter is unspecified, users of all states are included.
   UserState user_state_filter = 4;
-  // If the user_type_filter is unspecified, users of all types are included.
-  UserType user_type_filter = 5;
+  // If the user_type_filters is empty, users of all types are included.
+  repeated UserType user_type_filters = 5;
```
Comment (saquibmian) [inline proto/buf/alpha/registry/v1alpha1/user.proto:123]: Quick q: what happens if one of these is unspecified? Would we filter those out before we check for emptiness?

**Review by saquibmian — COMMENTED** [review]
Nice. We chatted offline, and while this does look incongruent with `user_state_filter`, it's probably better that this is an array and we only have to deal with migrating potentially one of them to not an an array.

File: proto/buf/alpha/registry/v1alpha1/user.proto (hunk 1 above)
Comment (cyinma) [reply proto/buf/alpha/registry/v1alpha1/user.proto:123]: it should be an invalid argument error IMO?

Comment (saquibmian) [reply proto/buf/alpha/registry/v1alpha1/user.proto:123]: Sounds good!

**Review by saquibmian — APPROVED** [review]

**Pull request merged**